Utils/Types/CircularBuffer.cpp

```diff
@@ -42,13 +47,16 @@ void CircularBuffer :: setup(U8* const buffer, const FwSizeType size) {
     FW_ASSERT(size > 0);
     FW_ASSERT(buffer != nullptr);
     FW_ASSERT(m_store == nullptr && m_store_size == 0); // Not already setup
+    FW_ASSERT(size < static_cast<FwSizeType>(static_cast<FwSizeType>(-1) / 2)); // Reasonable size limit
 
     // Initialize buffer data
     m_store = buffer;
     m_store_size = size;
     m_head_idx = 0;
     m_allocated_size = 0;
     m_high_water_mark = 0;
+    m_ser_offset = 0;
+    m_deser_offset = 0;
```

Comment:
Should `m_ser_offset` and `m_deser_offset` not be computed from `m_head_idx` and `m_allocated_size` ? Or is there a use case where they would differ?

If I do
```
circ.rotate(5);
circ.deserializeSkip(3);
```
will the `deserialize()` vs. `peek()` offsets not match?